### PR TITLE
Use dedicated dtype ModuleID and GroupName instead of string

### PR DESCRIPTION
**Pull request by mr0re1**
Rationale:
* To bolster interfaces;
* To enable move of validations to unmarshaling stage;

### CODE REVIEW

**Comment by tpdownes** [comment]
@mr0re1 #1262 created a merge conflict that needs to be addressed. #1265 (once merged) might, too.

File: pkg/config/config.go [hunk 1]
```diff
@@ -82,16 +83,30 @@ var movedModules = map[string]string{
 	"community/modules/scheduler/cloud-batch-login-node": "modules/scheduler/batch-login-node",
 }
 
+// GroupName is the name of a deployment group
+type GroupName string
+
+// Validate checks that the group name is valid
+func (n GroupName) Validate() error {
+	if n == "" {
+		return errors.New(errorMessages["emptyGroupName"])
```
Comment (nick-stroud) [inline pkg/config/config.go:92]: nit: Previously you had suggested we start moving these to constants. I would recommend we start using the new pattern as we touch these. Would you move this particular message to a constant? Same for "duplicateGroup" message in this PR.

File: pkg/config/config.go [hunk 2]
```diff
@@ -253,14 +268,17 @@ func (v *validatorConfig) check(name validatorName, requiredInputs []string) err
 	return nil
 }
 
+// ModuleID is a unique identifier for a module in a blueprint
+type ModuleID string
```
Comment (nick-stroud) [inline pkg/config/config.go:272]: nit: Should we validate this string is not empty?

File: pkg/config/config.go
Comment (nick-stroud) [inline pkg/config/config.go:1]: Make sure to rebase as Tom suggests.

**Review by nick-stroud — APPROVED** [review]

File: pkg/config/config.go (hunk 2 above)
Comment (mr0re1) [reply pkg/config/config.go:272]: We should, but it's out of scope

File: pkg/config/config.go (hunk 1 above)
Comment (mr0re1) [reply pkg/config/config.go:92]: Agree, but I would rather do it in separate change.

**Pull request merged**